components/aws/sagemaker/batch_transform/component.yaml

```diff
@@ -4,78 +4,102 @@ description: |
+    type: Integer
   - name: batch_strategy
     description: 'The number of records to include in a mini-batch for an HTTP inference request.'
     default: ''
+    type: String
   - name: environment
     description: 'The environment variables to set in the Docker container. Up to 16 key-value entries in the map.'
     default: '{}'
+    type: JSON
   - name: input_location
     description: 'The S3 location of the data source that is associated with a channel.'
+    type: String
   - name: data_type
     description: 'Data type of the input. Can be ManifestFile, S3Prefix, or AugmentedManifestFile.'
     default: 'S3Prefix'
+    type: String
   - name: content_type
     description: 'The multipurpose internet mail extension (MIME) type of the data.'
     default: ''
+    type: String
   - name: split_type
     description: 'The method to use to split the transform job data files into smaller batches.'
     default: 'None'
+    type: String
   - name: compression_type
     description: 'If the transform data is compressed, the specification of the compression type.'
     default: 'None'
+    type: String
   - name: output_location
     description: 'The Amazon S3 path where you want Amazon SageMaker to store the results of the transform job.'
+    type: String
   - name: accept
     description: 'The MIME type used to specify the output data.'
     default: ''
+    type: String
   - name: assemble_with
     description: 'Defines how to assemble the results of the transform job as a single S3 object. Either None or Line.'
     default: ''
+    type: String
   - name: output_encryption_key
     description: 'The AWS Key Management Service ID of the key used to encrypt the output data.'
     default: ''
+    type: String
   - name: input_filter
     description: 'A JSONPath expression used to select a portion of the input data to pass to the algorithm.'
     default: ''
+    type: String
   - name: output_filter
     description: 'A JSONPath expression used to select a portion of the joined dataset to save in the output file for a batch transform job.'
     default: ''
+    type: String
   - name: join_source
     description: 'Specifies the source of the data to join with the transformed data.'
     default: 'None'
+    type: String
   - name: instance_type
     description: 'The ML compute instance type.'
     default: 'ml.m4.xlarge'
+    type: String
   - name: instance_count
     description: 'The number of ML compute instances to use in each training job.'
     default: '1'
+    type: Integer
   - name: resource_encryption_key
     description: 'The AWS KMS key that Amazon SageMaker uses to encrypt data on the storage volume attached to the ML compute instance(s).'
     default: ''
+    type: String
   - name: endpoint_url
     description: 'The endpoint URL for the private link VPC endpoint.'
     default: ''
+    type: String
   - name: tags
     description: 'Key-value pairs to categorize AWS resources.'
     default: '{}'
+    type: JSON
```

Comment:
JFYI: The type name currently supported by the SDK is "JsonObject" or "Json". With those type names a dictionary argument will be converted to JSON string automatically.

Reply:
Is there a "JsonArray" type as well? Or would that be covered under "Json"?

Reply:
Ahh I found the list of types. Great. Thanks